paraatm/io/weather_cube_ET.py

```diff
@@ -0,0 +1,361 @@
+
+    Parameters
+    ----------
+    a: longitude in WGS84.
+
+    Returns
+    -------
+    longitude in Mercator
+    """
+    Radius = 6378137.0  # Radius of Earth
+    return math.radians(a) * Radius
+
+
+def find_nearest_value(array, num):
+    """
+    Find the nearest value to a given a number in a pre-defined array
+
+    Parameters
+    ----------
+    array: float, int
+    The reference array.
+
+    Returns
+    -------
+    num: float
+    The number we are interested.
+    """
+    nearest_val = array[abs(array - num) == abs(array - num).min()]
+    return nearest_val
+
+
+def check_convective_weather_files(weather_path, unix_time):
+    """
+    Find the closest ET weather files corresponding to the time of the current location
+
+    Parameters
+    ----------
+    weather_path: str
+    The path to the weather data.
+    unix_time: int
+    The unix time of the current location
+
+    Returns
+    -------
+    file_path: str
+    The path to the closest ET weather file at the current location and the current time
+    """
+    pin = datetime.datetime.utcfromtimestamp(int(float(unix_time))).strftime('%Y%m%d %H%M%S')  # time handle to check CIWS database
+    array = np.asarray([0, 230, 500, 730,
+                        1000, 1230, 1500, 1730,
+                        2000, 2230, 2500, 2730,
+                        3000, 3230, 3500, 3730,
+                        4000, 4230, 4500, 4730,
+                        5000, 5230, 5500, 5730])
+
+    nearest_value = int(find_nearest_value(array, 0.001+np.asarray([int(eliminate_zeros(pin[-4:]))])))  # find the closest time for downloading data from CIWS
+    nearest_value = make_up_zeros(str(nearest_value))  # make up zeros for 0 230 500 730
+
+    filename = pin[:8] + "ET/ciws.EchoTop." + pin[:8] + "T" + str(pin[-6:-4]) + nearest_value + "Z.nc"
+
+    file_path = weather_path + filename
+    return file_path
+
+
+def eliminate_zeros(num):
+    """
+    clear the zeros in the output of the unix to time transformation.
+
+    Parameters
+    ----------
+    num: float
+    A four digit number represents the minute and second time.
+
+    Returns
+    -------
+    num: float
+    The cleared last four digits time
+    """
+    if num[0] == '0' and num[1] == '0' and num[2] == '0':
+        return num[3]
+    if num[0] == '0' and num[1] == '0' and num[2] != '0':
+        return num[2:]
+    if num[0] == '0' and num[1] != '0':
+        return num[1:]
+    if num[0] != '0':
+        return num
+
+
+def make_up_zeros(str):
+    """
+    makeup the zeros to write the correct name of the weather ET file
+
+    Parameters
+    ----------
+    str: str
+    The objective string of numbers to makeup
+
+    Returns
+    -------
+    str: str
+    The string with four numbers
+    """
+    if len(str) == 4:
+        return str
+    if len(str) == 3:
+        return "0" + str
+    if len(str) == 2:
+        return "00" + str
+    if len(str) == 1:
+        return "000" + str

```

Comment:
How about just:
```suggestion
    '{:0>4}'.format(str)
```

Reply:
This is really a good idea. Thanks for making this such clear. Many thanks.

Reply:
Many thanks again. This helped.